### PR TITLE
feat: add icons for module and typescript support

### PR DESCRIPTION
**Pull request by juanpicado**
Display icons for:

- TypeScript support
- CommonJS or ES6 module

### CODE REVIEW

File: packages/plugins/ui-theme/types/packageMeta.ts [hunk 1]
```diff
@@ -25,6 +27,9 @@ export interface PackageMetaInterface {
       type?: string;
       url?: string;
     };
+    main?: string;
+    type?: ModuleType;
+    types?: string;
```
Comment (Splaktar) [inline packages/plugins/ui-theme/types/packageMeta.ts:32]: This is a nice feature for sure.

Question about how officially recognized the `types` field is?

I see it listed in https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html, but not in https://docs.npmjs.com/cli/v8/configuring-npm/package-json.

File: packages/plugins/ui-theme/src/components/Icons/DevsIcons/ES6Module.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import { Theme } from '@mui/material';
+import React from 'react';
+
+import icon from './es6modules.svg';
+
+const ImgIcon = styled.img<{ theme?: Theme }>(({ theme }) => ({
+  marginLeft: theme?.spacing(1),
+}));
+
+export function ES6Modules() {
```
Comment (Splaktar) [inline packages/plugins/ui-theme/src/components/Icons/DevsIcons/ES6Module.tsx:11]: In general, I would vote for a name change to `ESModules` and `ESM` in the SVG.

**Comment by Splaktar** [comment]
On the icon being `TS`, that might not always be the case. You can provide a `.d.ts` file in a pure JavaScript project and then still have them consumed by the IDE and JSDoc. I know it's longer, but perhaps having another way to display that the package has types might be more accurate.

File: packages/plugins/ui-theme/src/components/Icons/DevsIcons/ES6Module.tsx (hunk 2 above)
Comment (juanpicado) [reply packages/plugins/ui-theme/src/components/Icons/DevsIcons/ES6Module.tsx:11]: I'll rename it thanks !

**Comment by juanpicado** [comment]
> On the icon being `TS`, that might not always be the case. You can provide a `.d.ts` file in a pure JavaScript project and then still have them consumed by the IDE and JSDoc. I know it's longer, but perhaps having another way to display that the package has types might be more accurate.

You are right, I think I meant Types rather TypeScript.

File: packages/plugins/ui-theme/types/packageMeta.ts (hunk 1 above)
Comment (juanpicado) [reply packages/plugins/ui-theme/types/packageMeta.ts:32]: If there is `"types": "./index.d.ts",` on the package.json, yeah it is not an standard and seems open field, but well we can work with that, one day you will find slide of pizza within it and will be fine 🤷🏼‍♂️

**Pull request merged**